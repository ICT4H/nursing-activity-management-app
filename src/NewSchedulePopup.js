import React from "react";
import PersonDetails from './PersonDetails'
import SelectOptions from "./SelectOptions";
import SaveCancelButtons from "./SaveCancelButtons";
import {BID, CAP, MG, ML, QD, QID, QOD, TAB, TBSP, TID, TSP} from "./constants";
import Timings from "./Timings";

class NewSchedulePopup extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    //todo: Might come from server using an API
    this.medicineUnits = [TAB, CAP, ML, MG, TSP, TBSP];
    this.frequencies = [QD,BID,TID,QID,QOD];
  }

  handleChange() {
    let medicine = {
      medicineName: this.refs.medicineName.value,
      dose: this.refs.dose.value,
      unit: this.refs.unit.value,
      frequency:this.refs.frequency.value,
      startingDate:this.refs.startingDate.value
    };
    this.props.onChange(medicine);
  }

  render() {
    let medicine = this.props.medicine;
    return (
        <div className="popup">
          <div className="popupContent">
            <p>Add New Medicine Schedule</p>
            <PersonDetails person={this.props.patient}
                           className={"patientDetails"}/>
            <input
                type="text" placeholder="medicineName" ref="medicineName"
                value={medicine.medicineName} onChange={this.handleChange}/>

            <input
                type="text" placeholder="dose" ref="dose"
                value={medicine.dose} onChange={this.handleChange}/>

            <SelectOptions selectedValue={medicine.unit} ref="unit" options={this.medicineUnits}
                           className="chooseUnit" onChange={this.handleChange} chooseMsg="CHOOSE UNIT"/>

            <SelectOptions selectedValue={medicine.frequency} options={this.frequencies} className="chooseFrequency"
                           ref="frequency" chooseMsg="CHOOSE FREQUENCY" onChange={this.handleChange}/>

            <input type="date" ref="startingDate" placeholder="startingDate" onChange={this.handleChange} value={medicine.startingDate}/>
            <Timings timings={medicine.timings}/>

            <SaveCancelButtons saveFn={this.props.saveFn} cancelFn={this.props.cancelFn}/>
          </div>
        </div>
    )
  }
}

export default NewSchedulePopup;