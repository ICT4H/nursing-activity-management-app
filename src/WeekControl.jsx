import React, {Component} from 'react';
import moment from "moment/moment";
import PropTypes from "prop-types";
import WeeklyTable from "./WeeklyTable";

class WeekControl extends Component {
  render() {
    return (
        <div className={this.props.className}>
          <button onClick={this.props.pastWeek} className={"pastWeek"}>{"<"}</button>
          <p>{moment(this.props.currentWeek.startingDate).format('DD')}-
            {moment(this.props.currentWeek.endingDate).format('DD MMM')}
          </p>
          <button onClick={this.props.nextWeek} className={"nextWeek"}>{">"}</button>
        </div>
    )
  }
}

WeeklyTable.propTypes = {
  currentWeek: PropTypes.objectOf(PropTypes.instanceOf(Date)).isRequired,
  date: PropTypes.arrayOf(PropTypes.object)
};

export default WeekControl
