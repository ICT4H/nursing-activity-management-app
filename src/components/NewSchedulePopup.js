import React from "react";
import PersonDetails from './PersonDetails'
import SelectOptions from "./SelectOptions";
import SaveCancelButtons from "./SaveCancelButtons";
import {BID, CAP, MG, ML, QD, QID, QOD, TAB, TBSP, TID, TSP} from "../constants";
import {getResultantObject, mapFrequencyToNumber} from "../utils/utility";
import AdministrateTimes from "./AdministrateTimes";
import MedicationInput from "./MedicationInput";
import DosingInstructions from "./DosingInstructions";
import DosingPeriod from "./DosingPeriod";

class NewSchedulePopup extends React.Component {
  constructor(props) {
    super(props);
    this.handleMedicineNameChange = this.handleMedicineNameChange.bind(this);
    this.handleDoseChange = this.handleDoseChange.bind(this);
    this.handleFrequencyChange = this.handleFrequencyChange.bind(this);
    this.handleMedicineUnitChange = this.handleMedicineUnitChange.bind(this);
    this.handleStartingDateChange = this.handleStartingDateChange.bind(this);
    this.handleEndingDateChange = this.handleEndingDateChange.bind(this);
    //todo: Might come from server using an API
    this.medicineUnits = [TAB, CAP, ML, MG, TSP, TBSP];
    this.frequencies = [QD, BID, TID, QID, QOD];
  }

  render() {
    let drug = this.props.drug;
    return (
        <div className={this.getClassName()}>
          <div className="popupContent">
            <p>Add New Medicine Schedule</p>
            <PersonDetails person={this.props.patient} className={"patientDetails"}/>
            <MedicationInput drugName={drug.drugName} onChange={this.handleMedicineNameChange}/>
            <DosingInstructions medicineUnits={this.medicineUnits}
                                handleMedicineUnitChange={this.handleMedicineUnitChange}
                                doseUnit={drug.unit} doseValue={drug.dose}
                                handleDoseChange={this.handleDoseChange}
            />
            <SelectOptions selectedValue={drug.frequency} options={this.frequencies} className="chooseFrequency"
                           ref="frequency" chooseMsg="CHOOSE FREQUENCY" onChange={this.handleFrequencyChange}/>
            <DosingPeriod handleStartingDateChange={this.handleStartingDateChange}
                          handleEndingDateChange={this.handleEndingDateChange}
                          startingDate={drug.startingDate} endingDate={drug.endingDate}
            />
            <AdministrateTimes noOfTimeInputs={mapFrequencyToNumber(drug.frequency)}/>
            <SaveCancelButtons saveFn={this.props.saveFn} cancelFn={this.props.cancelFn}/>
          </div>
        </div>
    )
  }

  handleMedicineNameChange(event) {
    let changedDrugName = {drugName: event.target.value};
    let resultantDrug = getResultantObject(this.props.drug, changedDrugName);
    this.props.onChange(resultantDrug);
  }

  handleDoseChange(event) {
    let changedDose = {dose: event.target.value};
    let resultantDrug = getResultantObject(this.props.drug, changedDose);
    this.props.onChange(resultantDrug);
  }

  handleFrequencyChange(event) {
    let changedFrequency = {frequency: event.target.value};
    let resultantDrug = getResultantObject(this.props.drug, changedFrequency);
    this.props.onChange(resultantDrug);
  }

  handleStartingDateChange(event) {
    let startingDate = new Date(event.target.value);
    let changedStartingDate = {startingDate: startingDate};
    let resultantDrug = getResultantObject(this.props.drug, changedStartingDate);
    this.props.onChange(resultantDrug);
  }

  handleEndingDateChange(event) {
    let startingDate = new Date(event.target.value);
    let changedEndingDate = {endingDate: startingDate};
    let resultantDrug = getResultantObject(this.props.drug, changedEndingDate);
    this.props.onChange(resultantDrug);
  }

  getClassName() {
    return (this.props.hidden) ? "hidden" : "newSchedulePopup";
  }

  handleMedicineUnitChange(event) {
    let changedMedicineUnit = {unit: event.target.value};
    let resultantDrug = getResultantObject(this.props.drug, changedMedicineUnit);
    this.props.onChange(resultantDrug);
  }
}

export default NewSchedulePopup;
