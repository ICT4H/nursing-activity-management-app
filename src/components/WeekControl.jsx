import React, {Component} from 'react';
import moment from "moment/moment";
import PropTypes from "prop-types";

class WeekControl extends Component {
  render() {
    const currentWeek = this.props.currentWeek;
    return (
        <div className={this.props.className}>
          <button onClick={this.props.pastWeek} className={"pastWeek"}>{"<"}</button>
          <div>{moment(currentWeek.startingDate).format('DD')}-
            {moment(currentWeek.endingDate).format('DD MMM')}
          </div>
          <button onClick={this.props.nextWeek} className={"nextWeek"}>{">"}</button>
        </div>
    )
  }
}

WeekControl.propTypes = {
  currentWeek: PropTypes.objectOf(PropTypes.instanceOf(Date)).isRequired,
  date: PropTypes.arrayOf(PropTypes.object)
};

export default WeekControl
