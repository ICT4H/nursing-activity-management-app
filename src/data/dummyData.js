const patientDetails = {
  id: 'GAN12345',
  person: {
    name: "Cally Cardenas",
    gender: 'male',
    age: 33
  },
  patientUuid: "a81c9847-bfef-41f3-8bc2-8abfe11e3a6a"
};
let initialEmptyDrug = {
  drugName: "", dose: 1,
  unit: "",
  frequencyString: "",
  frequency: {name: "", uuid: "", frequencyPerDay: 0},
  startingDate: new Date('June 8, 2018 2:30:00'),
  endingDate: new Date('June 8, 2018 2:30:00')
};
export {patientDetails, initialEmptyDrug}
