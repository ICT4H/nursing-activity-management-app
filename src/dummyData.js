import Medicine from "./models/Medicine";
import {TOBEADMINISTRATED, ADMINISTRATED, NOTADMINISTRATED, TAB, ML, BID} from "./constants";

let sampleStartingDate = new Date("June 14, 2018 02:30:00");

let medicineData = {
  medicineName: "Paracetmol",
  dose: 1,
  unit: TAB,
  frequency: BID,
  startingDate: sampleStartingDate,
  noOfDays: 3,
  schedules: []
};


let medicineData2 = {
  medicineName: "Crosine",
  dose: 1,
  unit: TAB,
  frequency: BID,
  startingDate: sampleStartingDate,
  noOfDays: 3,
  schedules: []
};

const patientDetails = {
  id: 'GAN12345',
  name: "Cally Cardenas",
  gender: 'male',
  age: 33,
  medicinesToBeScheduled: [new Medicine(medicineData), new Medicine(medicineData2)],
};


let scheduledMedicineData = {
  medicineName: "Dopamine 40mg/ml",
  dose: 40,
  unit: ML,
  scheduleTimes:[new Date('June 5, 2018 2:30:00')],
  schedules: [{
    scheduledTime: new Date('June 5, 2018 9:30:00'),
    status: TOBEADMINISTRATED
  }, {
    scheduledTime: new Date('June 5, 2018 02:30:00'),
    status: NOTADMINISTRATED
  }, {
    scheduledTime: new Date('June 5, 2018 07:30:00'),
    status: NOTADMINISTRATED
  }, {
    scheduledTime: new Date('June 7, 2018 02:30:00'),
    status: ADMINISTRATED
  }, {
    scheduledTime: new Date('June 8, 2018 02:30:00'),
    status: TOBEADMINISTRATED
  }]
};

let scheduledMedicineData2 = {
  medicineName: "Metformin 850mg",
  dose: 1,
  unit: TAB,
  scheduleTimes:[new Date('June 8, 2018 2:30:00')],
  schedules: [{
    scheduledTime: new Date('June 2, 2018 02:30:00'),
    status: TOBEADMINISTRATED
  }, {
    scheduledTime: new Date('June 8, 2018 9:30:00'),
    status: NOTADMINISTRATED
  }, {
    scheduledTime: new Date('June 9, 2018 02:30:00'),
    status: NOTADMINISTRATED
  }, {
    scheduledTime: new Date('June 12, 2018 07:30:00'),
    status: NOTADMINISTRATED
  }, {
    scheduledTime: new Date('June 14, 2018 02:30:00'),
    status: ADMINISTRATED
  }]
};

let initialEmptyMedicine = {
  medicineName: "", dose: 0,
  startingDate:"",
  unit: "", dosage: "",frequency:""
};

export {patientDetails, scheduledMedicineData, scheduledMedicineData2, initialEmptyMedicine}