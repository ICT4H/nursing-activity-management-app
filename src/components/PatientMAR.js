import React from "react";
import ReactTable from "react-table";
import "react-table/react-table.css";
import NewSchedulePopup from './NewSchedulePopup';
import {initialEmptyMedicine} from "../Data/dummyData";
import {getSchedulesOf, groupByMedicineOrder} from "../utils/utility";
import WeekControl from "./WeekControl";
import Schedules from "./Schedules";
import MedicineDetails from "./MedicineDetails";
import PropTypes from "prop-types";
import DateUtils from "../utils/DateUtils";
import Headers from "./Headers";
import 'whatwg-fetch';

class PatientMAR extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentMedicineToPopup: initialEmptyMedicine,
      shallHidePopup: true,
      today: props.today,
      patientUuid: props.patientUuid,
      currentWeekData: []
    };
    this.showNewSchedulePopup = this.showNewSchedulePopup.bind(this);
    this.updateCurrentMedicine = this.updateCurrentMedicine.bind(this);
    this.goToPastWeek = this.goToPastWeek.bind(this);
    this.goToNextWeek = this.goToNextWeek.bind(this);
    this.getThisWeekData = this.getThisWeekData.bind(this);
    this.saveFn = this.saveFn.bind(this);
    this.hidePopup = this.hidePopup.bind(this);
  }

  componentWillMount() {
    this.getThisWeekData();
    let currentWeek = DateUtils.getWeekStatingAndEndingDates(this.props.today);
    this.setState({currentWeek: currentWeek})
  }

  showNewSchedulePopup(currentMedicineToPopup) {
    this.setState({shallHidePopup: false});
    this.updateCurrentMedicine(currentMedicineToPopup);
  }

  updateCurrentMedicine(medicine) {
    this.setState({currentMedicineToPopup: medicine});
  }

  goToPastWeek() {
    let pastWeek = DateUtils.getPastWeekStatingAndEndingDates(this.state.currentWeek.startingDate);
    this.setState({currentWeek: pastWeek});
  }

  goToNextWeek() {
    let nextWeek = DateUtils.getNextWeekStatingAndEndingDates(this.state.currentWeek.startingDate);
    this.setState({currentWeek: nextWeek});
  }

  getThisWeekData() {
    fetch(`/openmrs/ws/rest/v1/ipd/schedules/patient/${this.state.patientUuid}`, {credentials: "include"}).then((res) => {
      res.json().then((schedules) => {
        let currentWeekData = groupByMedicineOrder(schedules);
        this.setState({currentWeekData});
      });
    }).catch((err) => {
      // console.log(err);
    });
  }

  hidePopup() {
    this.updateCurrentMedicine(initialEmptyMedicine);
    this.setState({shallHidePopup: true})
  }

  saveFn() {
    //todo: Save the medicines schedules
    this.hidePopup();
  }

  getCurrentWeekColumns() {
    let startingDate = this.state.currentWeek.startingDate;
    let endingDate = this.state.currentWeek.endingDate;
    let columns = [];
    let medicineDetailColumn = {
      Header: "MedicineDetails",
      id: "MedicineDetails",
      accessor: d => <MedicineDetails medicine={d} onClick={this.showNewSchedulePopup.bind(null, d)}/>
    };
    columns.push(medicineDetailColumn);
    for (let date = startingDate; DateUtils.isInBetween(date, startingDate, endingDate); date = DateUtils.addDays(date, 1)) {
      let formattedDate = DateUtils.getDateAndDay(date);
      let column = {
        Header: formattedDate,
        id: formattedDate,
        accessor: d => <Schedules schedules={getSchedulesOf(date, d.schedules)}/>
      };
      columns.push(column);
    }
    return columns;
  }

  render() {
    let patient = this.props.patient;
    const currentWeek = this.state.currentWeek;
    return (
        <div>
          <NewSchedulePopup medicine={this.state.currentMedicineToPopup} hidden={this.state.shallHidePopup}
                            patient={patient} onChange={this.updateCurrentMedicine}
                            saveFn={this.saveFn} cancelFn={this.hidePopup}
          />

          <Headers patient={patient} today={this.state.today}
                   onClickOfButton={this.showNewSchedulePopup.bind(this, this.state.currentMedicineToPopup)}/>
          <ReactTable
              data={this.state.currentWeekData}
              columns={this.getCurrentWeekColumns()}
              className="-highlight"
              PaginationComponent={() => <WeekControl className="weekControl" goToPastWeek={this.goToPastWeek}
                                                      goToNextWeek={this.goToNextWeek} currentWeek={currentWeek}
              />}
              minRows={this.state.currentWeekData.length}
              showPaginationTop={true}
              showPaginationBottom={false}
          />
        </div>
    )
  }
}

PatientMAR.propTypes = {
  today: PropTypes.instanceOf(Date).isRequired
};


export default PatientMAR;
