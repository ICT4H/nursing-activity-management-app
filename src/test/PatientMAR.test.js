import React from 'react';
import PatientMAR from '../PatientMAR';
import renderer from 'react-test-renderer';
import {patientDetails} from "../dummyData";
import NewSchedulePopup from "../NewSchedulePopup";
import WeekControl from "../WeekControl";
import WeeklyTable from "../WeeklyTable";

test('Should have NewSchedulePopup with having patient as property', () => {
  const component = renderer.create(
      <PatientMAR patient={patientDetails}/>,
  );
  let tree = component.toJSON();
  expect(component.root.findByType(NewSchedulePopup).props.patient).toBe(patientDetails);
  expect(tree).toMatchSnapshot();
});

test('Should have WeekControl component with having property currentWeek', () => {
  const component = renderer.create(
      <PatientMAR patient={patientDetails}/>,
  );
  let tree = component.toJSON();
  expect(component.root.findByType(WeekControl).props.currentWeek);
  expect(tree).toMatchSnapshot();
});

test('Should have WeeklyTable component with having property currentWeek and currentWeekData', () => {
  const component = renderer.create(
      <PatientMAR patient={patientDetails}/>,
  );
  let tree = component.toJSON();
  expect(component.root.findByType(WeeklyTable).props.currentWeek);
  expect(component.root.findByType(WeeklyTable).props.currentWeekData);
  expect(tree).toMatchSnapshot();
});