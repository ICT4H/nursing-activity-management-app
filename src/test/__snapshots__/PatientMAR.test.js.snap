// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have NewSchedulePopup with having patient as property 1`] = `
<div>
  <div
    className="popup"
  >
    <div
      className="popupContent"
    >
      <p>
        Add New Medicine Schedule
      </p>
      <div
        className="patientDetails"
      >
        <p>
          Cally Cardenas
        </p>
        <p>
          male
        </p>
        <p>
          33 yrs
        </p>
      </div>
      <input
        onChange={[Function]}
        placeholder="medicineName"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="dose"
        type="text"
        value={0}
      />
      <select
        className="chooseUnit"
        onChange={[Function]}
        value=""
      >
        <option>
          CHOOSE UNIT
        </option>
        <option
          value="TABLETS"
        >
          TABLETS
        </option>
        <option
          value="CAPSULES"
        >
          CAPSULES
        </option>
        <option
          value="ML"
        >
          ML
        </option>
        <option
          value="MG"
        >
          MG
        </option>
        <option
          value="TEASPOON"
        >
          TEASPOON
        </option>
        <option
          value="TABLESPOON"
        >
          TABLESPOON
        </option>
      </select>
      <select
        className="chooseFrequency"
        onChange={[Function]}
        value={undefined}
      >
        <option>
          CHOOSE FREQUENCY
        </option>
        <option
          value="ONCE A DAY"
        >
          ONCE A DAY
        </option>
        <option
          value="TWICE A DAY"
        >
          TWICE A DAY
        </option>
        <option
          value="THRICE A DAY"
        >
          THRICE A DAY
        </option>
        <option
          value="FOUR TIMES A DAY"
        >
          FOUR TIMES A DAY
        </option>
        <option
          value="ON ALTERNATE DAYS"
        >
          ON ALTERNATE DAYS
        </option>
      </select>
      <input
        onChange={[Function]}
        placeholder="startingDate"
        type="date"
        value=""
      />
      <div>
        <button
          onClick={[Function]}
        >
          + Add More Times
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          value="Save"
        >
          Save
        </button>
        <button
          className="button"
          onClick={[Function]}
          value="Cancel"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
  <div
    className="titles"
  >
    <div
      className="patientDetails"
    >
      <p>
        Cally Cardenas
      </p>
      <p>
        male
      </p>
      <p>
        33 yrs
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      + add more
    </button>
    <p>
      Thu Jun 14 2018
    </p>
  </div>
  <div
    className="changeWeek"
  >
    <button
      className="pastWeek"
      onClick={[Function]}
    >
      &lt;
    </button>
    <p>
      10
      -
      16 Jun
    </p>
    <button
      className="nextWeek"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            medicineName
          </th>
          <th>
            Sun Jun 10 2018
          </th>
          <th>
            Mon Jun 11 2018
          </th>
          <th>
            Tue Jun 12 2018
          </th>
          <th>
            Wed Jun 13 2018
          </th>
          <th>
            Thu Jun 14 2018
          </th>
          <th>
            Fri Jun 15 2018
          </th>
          <th>
            Sat Jun 16 2018
          </th>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Dopamine 40mg/ml
              </p>
              <p>
                40
                 
                ML
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Metformin 850mg
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div>
              <div
                className="schedule"
              >
                <p>
                  7:30
                </p>
                 
                <span>
                  ⚠
                </span>
              </div>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div>
              <div
                className="schedule"
              >
                <p>
                  2:30
                </p>
                 
                <span>
                  ✅
                </span>
              </div>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Paracetmol
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Crosine
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Should have WeekControl component with having property currentWeek 1`] = `
<div>
  <div
    className="popup"
  >
    <div
      className="popupContent"
    >
      <p>
        Add New Medicine Schedule
      </p>
      <div
        className="patientDetails"
      >
        <p>
          Cally Cardenas
        </p>
        <p>
          male
        </p>
        <p>
          33 yrs
        </p>
      </div>
      <input
        onChange={[Function]}
        placeholder="medicineName"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="dose"
        type="text"
        value={0}
      />
      <select
        className="chooseUnit"
        onChange={[Function]}
        value=""
      >
        <option>
          CHOOSE UNIT
        </option>
        <option
          value="TABLETS"
        >
          TABLETS
        </option>
        <option
          value="CAPSULES"
        >
          CAPSULES
        </option>
        <option
          value="ML"
        >
          ML
        </option>
        <option
          value="MG"
        >
          MG
        </option>
        <option
          value="TEASPOON"
        >
          TEASPOON
        </option>
        <option
          value="TABLESPOON"
        >
          TABLESPOON
        </option>
      </select>
      <select
        className="chooseFrequency"
        onChange={[Function]}
        value={undefined}
      >
        <option>
          CHOOSE FREQUENCY
        </option>
        <option
          value="ONCE A DAY"
        >
          ONCE A DAY
        </option>
        <option
          value="TWICE A DAY"
        >
          TWICE A DAY
        </option>
        <option
          value="THRICE A DAY"
        >
          THRICE A DAY
        </option>
        <option
          value="FOUR TIMES A DAY"
        >
          FOUR TIMES A DAY
        </option>
        <option
          value="ON ALTERNATE DAYS"
        >
          ON ALTERNATE DAYS
        </option>
      </select>
      <input
        onChange={[Function]}
        placeholder="startingDate"
        type="date"
        value=""
      />
      <div>
        <button
          onClick={[Function]}
        >
          + Add More Times
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          value="Save"
        >
          Save
        </button>
        <button
          className="button"
          onClick={[Function]}
          value="Cancel"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
  <div
    className="titles"
  >
    <div
      className="patientDetails"
    >
      <p>
        Cally Cardenas
      </p>
      <p>
        male
      </p>
      <p>
        33 yrs
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      + add more
    </button>
    <p>
      Thu Jun 14 2018
    </p>
  </div>
  <div
    className="changeWeek"
  >
    <button
      className="pastWeek"
      onClick={[Function]}
    >
      &lt;
    </button>
    <p>
      10
      -
      16 Jun
    </p>
    <button
      className="nextWeek"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            medicineName
          </th>
          <th>
            Sun Jun 10 2018
          </th>
          <th>
            Mon Jun 11 2018
          </th>
          <th>
            Tue Jun 12 2018
          </th>
          <th>
            Wed Jun 13 2018
          </th>
          <th>
            Thu Jun 14 2018
          </th>
          <th>
            Fri Jun 15 2018
          </th>
          <th>
            Sat Jun 16 2018
          </th>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Dopamine 40mg/ml
              </p>
              <p>
                40
                 
                ML
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Metformin 850mg
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div>
              <div
                className="schedule"
              >
                <p>
                  7:30
                </p>
                 
                <span>
                  ⚠
                </span>
              </div>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div>
              <div
                className="schedule"
              >
                <p>
                  2:30
                </p>
                 
                <span>
                  ✅
                </span>
              </div>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Paracetmol
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Crosine
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Should have WeeklyTable component with having property currentWeek and currentWeekData 1`] = `
<div>
  <div
    className="popup"
  >
    <div
      className="popupContent"
    >
      <p>
        Add New Medicine Schedule
      </p>
      <div
        className="patientDetails"
      >
        <p>
          Cally Cardenas
        </p>
        <p>
          male
        </p>
        <p>
          33 yrs
        </p>
      </div>
      <input
        onChange={[Function]}
        placeholder="medicineName"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="dose"
        type="text"
        value={0}
      />
      <select
        className="chooseUnit"
        onChange={[Function]}
        value=""
      >
        <option>
          CHOOSE UNIT
        </option>
        <option
          value="TABLETS"
        >
          TABLETS
        </option>
        <option
          value="CAPSULES"
        >
          CAPSULES
        </option>
        <option
          value="ML"
        >
          ML
        </option>
        <option
          value="MG"
        >
          MG
        </option>
        <option
          value="TEASPOON"
        >
          TEASPOON
        </option>
        <option
          value="TABLESPOON"
        >
          TABLESPOON
        </option>
      </select>
      <select
        className="chooseFrequency"
        onChange={[Function]}
        value={undefined}
      >
        <option>
          CHOOSE FREQUENCY
        </option>
        <option
          value="ONCE A DAY"
        >
          ONCE A DAY
        </option>
        <option
          value="TWICE A DAY"
        >
          TWICE A DAY
        </option>
        <option
          value="THRICE A DAY"
        >
          THRICE A DAY
        </option>
        <option
          value="FOUR TIMES A DAY"
        >
          FOUR TIMES A DAY
        </option>
        <option
          value="ON ALTERNATE DAYS"
        >
          ON ALTERNATE DAYS
        </option>
      </select>
      <input
        onChange={[Function]}
        placeholder="startingDate"
        type="date"
        value=""
      />
      <div>
        <button
          onClick={[Function]}
        >
          + Add More Times
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          value="Save"
        >
          Save
        </button>
        <button
          className="button"
          onClick={[Function]}
          value="Cancel"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
  <div
    className="titles"
  >
    <div
      className="patientDetails"
    >
      <p>
        Cally Cardenas
      </p>
      <p>
        male
      </p>
      <p>
        33 yrs
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      + add more
    </button>
    <p>
      Thu Jun 14 2018
    </p>
  </div>
  <div
    className="changeWeek"
  >
    <button
      className="pastWeek"
      onClick={[Function]}
    >
      &lt;
    </button>
    <p>
      10
      -
      16 Jun
    </p>
    <button
      className="nextWeek"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            medicineName
          </th>
          <th>
            Sun Jun 10 2018
          </th>
          <th>
            Mon Jun 11 2018
          </th>
          <th>
            Tue Jun 12 2018
          </th>
          <th>
            Wed Jun 13 2018
          </th>
          <th>
            Thu Jun 14 2018
          </th>
          <th>
            Fri Jun 15 2018
          </th>
          <th>
            Sat Jun 16 2018
          </th>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Dopamine 40mg/ml
              </p>
              <p>
                40
                 
                ML
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Metformin 850mg
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div>
              <div
                className="schedule"
              >
                <p>
                  7:30
                </p>
                 
                <span>
                  ⚠
                </span>
              </div>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div>
              <div
                className="schedule"
              >
                <p>
                  2:30
                </p>
                 
                <span>
                  ✅
                </span>
              </div>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Paracetmol
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="medicineDetails"
              onClick={[Function]}
            >
              <p>
                Crosine
              </p>
              <p>
                1
                 
                TABLETS
              </p>
            </div>
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
          <td>
            <div />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
